docs: adding documentation for releasing on a Jenkins instance (#324)
relekang/python-semantic-release: docs/automatic-releases/index.rst

The branch check, checks against the branch that gitlab-ci said it checked out, not the current
branch.

Jenkins
^^^^^^^
*Condition:* Environment variable ``JENKINS_URL`` is set.

> Determines the branch name from if either the environment variable ``BRANCH_NAME`` or the environment variable ``GIT_BRANCH`` for the build.
> Checks to ensure that the build is on the correct branch. Also, if ``CHANGE_ID`` is set, meaning it
> is a PR from a multibranch pipeline the build will not be automatically released.

Reviewer: ```suggestion
Determines the branch name from either the environment variable ``BRANCH_NAME``
or the environment variable ``GIT_BRANCH``, and checks to ensure that the build is on
the correct branch. Also, if ``CHANGE_ID`` is set, meaning it is a PR from a multi-branch
pipeline, the build will not be automatically released.
```
